### ISSUE TITLE
[PM-1765] for SubWorkflow jobs, the planner argument, --output-sites, isn't being set

### ISSUE THREAD

**Issue by mayani** [pegasus-isi/pegasus#1879]
When running the hierarchical workflow in the attached submit directory, I had not set --output-sites in the SubWorkflow job, which caused the workflow to fail. The attached screenshot illustrates the workflow I was trying to run. Relates to #1880

**Reporter:** @ryantanaka
**Resolution:** Fixed
**Watchers:**
@ryantanaka
@vahi
**Attachments:**

**Comment by mayani** [comment]
**Author:** @ryantanaka

I've added the workflow generation script, `workflow.py`.

**Comment by mayani** [comment]
**Author:** @vahi

the other case here that also is to consider, is that if no output site is specified, then the cache file for the SubWorkflow1 does not log the location of the outputs, k1.txt that is required for the SubWorkflow2 sub workflow

**Comment by mayani** [comment]
**Author:** @vahi

This is now fixed, and bamboo test 045-hierarchy-sharedfs-d tests this case in addition to #1880

As part of this we also simplified in planner as to how file locations of files generated are logged in the cache files. Even if staging outputs, locations for generated output files are now tracked in the function trackCaches uniformly.  This allows us to handle various cases of no output staging, staging to user specifed output site, staging to output map file location in one place.  Earlier, in case where the planner was staging outputs the locations were put in the cache files as we were determining the associated file transfers of output files. This was done to prevent some duplicate computations of the URL's.
